usaspending_api/etl/award_helpers.py

```diff
@@ -119,3 +120,59 @@ def update_contract_awards(award_tuple=None):
         rows = cursor.rowcount
 
     return rows
+
+
+def get_award_financial_transaction(
+        toptier_agency_cgac, piid=None, parent_award_id=None, fain=None, uri=None):
+    """
+    For specified award financial (aka "File C") data, try to find a matching
+    transaction (aka "File D"). We sometimes need to do this  because File C
+    doesn't always have the level of award/transaction specificity that we
+    want, so we try to find a matching File D record to grab the additional
+    information.
+
+    For example, when trying to match award financial information to an
+    award record, we need the awarding subtier agency, which isn't supplied
+    on File C. Thus, we'll use this function to find a File D record and
+    use the subtier agency information supplied there.
+
+    If we find more than one match, return the record with this most
+    recent action date.
+
+    Args:
+        toptier_agency_cgac: top tier agency code (aka CGAC code) from File C
+        piid: piid from File C (contract awards only)
+        parent_award_id: parent award id from File C (contract awards only)
+        fain: fain from File C (assistance awards only)
+        uri: uri from File C (assistance awards only)
+
+    Returns:
+        A Transaction model instance
+    """
+    # if both fain and uri are supplied as paramaters, look up by fain first
+    incoming_fain = fain
+    incoming_uri = uri
+    if incoming_fain is not None and incoming_uri is not None:
+        uri = None
```

Comment:
Right, I think we just don't want `assistance_data__uri` in the filter _at all_, rather than checking for None. Unless we actually do want to only find awards where the FAIN matches but URI is none.